### ISSUE TITLE
Walkaround needs an "Import All" button

### ISSUE THREAD

**Issue by GoogleCodeExporter** [strikersix23/walkaround#13]
```
Importing each wave individually would take a very long time. It would be good 
if there were a button to import all of them, so we could leave it running 
overnight. Thanks,
```

Original issue reported on code.google.com by `tal...@gmail.com` on 24 Nov 2011 at 10:10

**Comment by GoogleCodeExporter** [comment]
```
I think each wave should have a checkbox on right of it, and on the bottom we 
should have 'Import selected waves'. There should be also 'Select all' / 
'Unselect all' waves. This way I can easly import for example all waves but 
ignore few ones I do not really need imported.

```

Original comment by `witold.b...@gmail.com` on 24 Nov 2011 at 4:01

**Comment by GoogleCodeExporter** [comment]
Original comment by `oh...@google.com` on 24 Nov 2011 at 10:07
- Changed state: **Accepted**
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
As server owner, I don't really like that option :D. It gonna eat a lot of CPU 
time. I would prefer that people would be able to search for specific wave, or 
import by wave id, so the CPU resources are spent only on important waves. For 
example, I have thousands of junk waves - as a result of a runaway robot (that 
I created and tested on my account). No point to import all these waves.
```

Original comment by `vega113` on 30 Nov 2011 at 9:49

**Comment by GoogleCodeExporter** [comment]
```
+1 for checkboxes, search operators or waveID selections. Users should sort and 
decide which waves they want to import. Right now the import function is 
creating a mess in the Walkaround inbox.
```

Original comment by `jeremy.ngl` on 30 Nov 2011 at 10:47

**Comment by GoogleCodeExporter** [comment]
```
I guess that "Import All" should be disabled on the demo servers, but our 
organisation would be running our own instance. We want to copy all our data 
off wave.google.com before it is shut down. The plan is that each person logs 
in and imports all their waves... I assume it's smart enough to avoid 
duplicating waves that someone else already imported?
```

Original comment by `tal...@gmail.com` on 1 Dec 2011 at 10:20

**Comment by GoogleCodeExporter** [comment]
```
I agree that there should be a simple way to import all waves, perhaps with an 
admin option to disable it.  If any of you want to implement sorting, 
searching, and checkboxes (is there a JS library that would make this really 
easy?), patches welcome.

The server side also needs a few improvements for "import all" to make sense, 
such as importing large waves without errors, importing attachments, and -- as 
you say -- avoiding duplicate imports (see also issue 2).

```

Original comment by `oh...@google.com` on 1 Dec 2011 at 7:16

**Comment by GoogleCodeExporter** [comment]
Original comment by `oh...@google.com` on 4 Jan 2012 at 12:35
- Added labels: **Component-Import**

**Comment by GoogleCodeExporter** [comment]
Original comment by `oh...@google.com` on 19 Jan 2012 at 8:25

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 50ac31a67976.
```

Original comment by `oh...@google.com` on 2 Feb 2012 at 10:38
- Changed state: **Fixed**
